src/database/setup.py

```diff
@@ -44,23 +43,44 @@ def connect_to_database(
     return engine
 
 
-def populate_database(engine: Engine, only_if_empty: bool = True, data: str = "example"):
-    """Adds some data to the Dataset and Publication tables.
+def populate_database(
+    engine: Engine,
+    only_if_empty: bool = True,
+    platform_data: str = "example",
+    platform_publications="example",
+):
+    """Add some data to the Dataset and Publication tables.
 
-    data: str (default="example")
-        One of "example" or "openml".
+    platform_data: str (default="example"): One of "nothing", "example", "huggingface" or "openml".
+    platform_publications: str (default="example"): One of "nothing" or "example".
     """
-    match data:
-        case "example":
-            datasets, publications = get_example_data()
-        case "openml":
-            datasets, publications = get_openml_data()
-        case _:
-            raise ValueError(f"`data` must be one of 'example' or 'openml': {data=}")
 
+    if platform_data == "nothing":
+        datasets = []
+    else:
+        dataset_connector = connectors.dataset_connectors.get(Platform(platform_data), None)
+        if dataset_connector is None:
+            possibilities = ", ".join(f"`{c}`" for c in connectors.dataset_connectors.keys())
+            msg = f"{platform_data=}, but must be one of {possibilities}"
+            raise NotImplementedError(msg)
```

Comment:
In my opinion, invalid values should raise a `ValueError`, even in this case where an invalid value might just be an unsupported platform.
Also, I think it would be better to do this input checking at the start of the function. Otherwise you might end up downloading all OpenML datasets only to get an error because you made a typo for the `platform_publications` parameter.